src/components/IconLibrary/icon-library.scss

```
@@ -47,53 +77,122 @@
   }
 }
 
-.icon-search {
-  margin-bottom: rem(32px);
+.icons-list {
+  display: flex;
+  flex-flow: wrap;
+  width: 100%;
+  padding-bottom: $spacing-06;
 }
 
-.icon {
+.icon__container {
+  flex: 0 1 50%;
+  position: relative;
+
   &:nth-child(n) {
-    border-left: 1px solid $carbon--gray-20;
     border-top: 1px solid $carbon--gray-20;
+    border-left: 1px solid $carbon--gray-20;
   }
-
-  &:nth-child(-n + 4) {
+  &:nth-child(n):nth-child(-n + 2) {
     border-top: none;
   }
-
-  &:nth-child(4n + 1) {
+  &:nth-child(2n - 1) {
     border-left: none;
   }
 
-  @include carbon--breakpoint('lg') {
+  & .bx--aspect-ratio > * {
+    z-index: 100; // all siblings of bx--aspect-ratio--object will have same z-index
+  }
+}
+
+@media all and (min-width: 515px) {
+  .icon__container {
+    flex: 0 1 calc(100% / 3);
+
     &:nth-child(n) {
+      border-top: 1px solid $carbon--gray-20;
       border-left: 1px solid $carbon--gray-20;
+    }
+    &:nth-child(n):nth-child(-n + 3) {
+      border-top: none;
+    }
+    &:nth-child(3n - 2) {
+      border-left: none;
+    }
+  }
+}
+
+@media all and (min-width: 730px) {
+  .icon__container {
+    flex: 0 1 25%;
+
+    &:nth-child(n) {
       border-top: 1px solid $carbon--gray-20;
+      border-left: 1px solid $carbon--gray-20;
     }
+    &:nth-child(n):nth-child(-n + 4) {
+      border-top: none;
+    }
+    &:nth-child(4n - 3) {
+      border-left: none;
+    }
+  }
+}
 
-    &:nth-child(-n + 6) {
+@media all and (min-width: 980px) {
+  .icon__container {
+    flex: 0 1 20%;
+
+    &:nth-child(n) {
+      border-top: 1px solid $carbon--gray-20;
+      border-left: 1px solid $carbon--gray-20;
+    }
+    &:nth-child(n):nth-child(-n + 5) {
       border-top: none;
     }
+    &:nth-child(5n - 4) {
+      border-left: none;
+    }
+  }
+}
 
-    &:nth-child(6n + 1) {
+@media all and (min-width: 1300px) {
+  .icon__container {
+    flex: 0 1 calc(100% / 6);
+
+    &:nth-child(n) {
+      border-top: 1px solid $carbon--gray-20;
+      border-left: 1px solid $carbon--gray-20;
+    }
+    &:nth-child(n):nth-child(-n + 6) {
+      border-top: none;
+    }
+    &:nth-child(6n - 5) {
       border-left: none;
     }
   }
 }
 
 .icon__card {
   background: $ui-02;
-  height: 100%;
   text-align: center;
-  position: relative;
   display: flex;
   align-items: center;
   justify-content: center;
+
+  &:hover,
+  &:focus,
+  .icon__actions:focus,
+  button:focus,
+  a:focus {
```

```suggestion
  a:focus .icon__actions {
```

does this need to be nested?